src/main/java/net/dzikoysk/funnyguilds/basic/user/UserManager.java

```
@@ -0,0 +1,168 @@
+        return Option.of(usersByUuid.get(uuid));
+    }
+
+    public Option<User> getUser(Player player) {
+        if (player.getUniqueId().version() == 2) {
+            return Option.of(new User(player));
+        }
+
+        return getUser(player.getUniqueId());
+    }
+
+    public Option<User> getUser(OfflinePlayer offline) {
+        return getUser(offline.getName());
+    }
+
+    public User create(UUID uuid, String name) {
+        Validate.notNull(uuid, "uuid can't be null!");
+        Validate.notNull(name, "name can't be null!");
+        Validate.notBlank(name, "name can't be blank!");
+        Validate.isTrue(UserUtils.validateUsername(name), "name is not valid!");
+
+        User user = new User(uuid, name);
+
+        addUser(user);
+        RankManager.getInstance().update(user);
+
+        return user;
+    }
+
+    public User create(Player player) {
+        Validate.notNull(player, "player can't be null!");
+
+        User user = new User(player);
+        addUser(user);
+        RankManager.getInstance().update(user);
+
+        return user;
+    }
+
+    public void addUser(User user) {
+        Validate.notNull(user, "user can't be null!");
+
+        usersByUuid.put(user.getUUID(), user);
+        usersByName.put(user.getName(), user);
+    }
+
+    public void removeUser(User user) {
+        Validate.notNull(user, "user can't be null!");
+
+        usersByUuid.remove(user.getUUID());
+        usersByName.remove(user.getName());
+    }
+
+    public void updateUsername(User user, String newUsername) {
+        Validate.notNull(user, "user can't be null!");
+
+        usersByName.remove(user.getName());
+        usersByName.put(newUsername, user);
+
+        user.setName(newUsername);
+    }
+
+    public boolean playedBefore(String nickname) {
+        return playedBefore(nickname, false);
+    }
+
+    public boolean playedBefore(String nickname, boolean ignoreCase) {
+        if (nickname == null) {
+            return false;
```

Maybe throw an exception with Preconditions & StringUtils.isEmpty()?